### ISSUE TITLE
s6-svc zombie processes on each reload

### ISSUE THREAD

**Issue by Frankkkkk** [haproxytech/kubernetes-ingress#428]
Hi,
on `haproxytech/kubernetes-ingress:1.7.5`, a new zombie process is spawned on each `Ingress` change.

This leads to o huge number of zombie processes and sometimes a connectivity loss (not yet debugged sadly).

Example:
```
root        6235  0.0  0.0  80528  1892 ?        Ssl  Feb23   0:02 /usr/bin/conmon -b /run/containers/storage/overlay-containers/5bd27e1bca9b6b81731ec090d76ae832b6119f431f8139c05f23c3ef2f544e71/userdata -c 5bd27e1bca9b6b81731ec090d76ae832b6119f431f8139c05f23c3ef2f544e71 --exit-dir /var/run/crio/exits -l /var/log/pods/haproxy-controller_haproxy-ingress-5f44dcddd-jw7bw_87e2eb18-410e-4bfa-99d0-5630cc9b167d/haproxy-ingress/0
ubuntu      6249  0.0  0.0    204     4 ?        Ss   Feb23   0:00  \_ s6-svscan -t0 /var/run/s6/services
ubuntu      6305  0.0  0.0    204     4 ?        S    Feb23   0:00      \_ s6-supervise s6-fdholderd
ubuntu      6474  0.0  0.0    204     4 ?        S    Feb23   0:00      \_ s6-supervise haproxy
ubuntu      6726  0.0  0.3 124596 88420 ?        Ss   Feb23   1:20      |   \_ /usr/local/sbin/haproxy -sf 14873 14847 -x /var/run/haproxy-runtime-api.sock -W -db -f /etc/haproxy/haproxy.cfg -f /etc/haproxy/haproxy-aux.cfg
ubuntu    857610  2.6  0.3 147568 81616 ?        Sl   09:12   0:03      |       \_ /usr/local/sbin/haproxy -sf 14873 14847 -x /var/run/haproxy-runtime-api.sock -W -db -f /etc/haproxy/haproxy.cfg -f /etc/haproxy/haproxy-aux.cfg
ubuntu      6475  0.0  0.0    204     4 ?        S    Feb23   0:00      \_ s6-supervise ingress-controller
ubuntu      6478  0.1  0.4 885536 111164 ?       Ssl  Feb23  40:27          \_ /haproxy-ingress-controller --with-s6-overlay --configmap=haproxy-controller/haproxy --default-backend-service=haproxy-controller/ingress-default-backend
ubuntu      6711  0.0  0.0      0     0 ?        Z    Feb23   0:00              \_ [s6-svc] <defunct>
ubuntu      8258  0.0  0.0      0     0 ?        Z    Feb23   0:00              \_ [s6-svc] <defunct>
ubuntu      8282  0.0  0.0      0     0 ?        Z    Feb23   0:00              \_ [s6-svc] <defunct>
ubuntu     53166  0.0  0.0      0     0 ?        Z    Feb24   0:00              \_ [s6-svc] <defunct>
ubuntu     53455  0.0  0.0      0     0 ?        Z    Feb24   0:00              \_ [s6-svc] <defunct>
ubuntu     53662  0.0  0.0      0     0 ?        Z    Feb24   0:00              \_ [s6-svc] <defunct>
ubuntu     53682  0.0  0.0      0     0 ?        Z    Feb24   0:00              \_ [s6-svc] <defunct>
--- snip ---
```

```
ps -faux | grep s6-svc | wc -l
738
```
and after creating a new ingress:
```
ps -faux | grep s6-svc | wc -l
739
```

The (stripped) pod container spec is: 
```yaml
  containers:
  - args:
    - --configmap=haproxy-controller/haproxy
    - --default-backend-service=haproxy-controller/ingress-default-backend
    env:
    - name: TZ
      value: Etc/UTC
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    image: haproxytech/kubernetes-ingress:1.7.5
    imagePullPolicy: Always
    name: haproxy-ingress
    resources:
      requests:
        cpu: 500m
        memory: 50Mi
    securityContext:
      capabilities:
        add:
        - NET_BIND_SERVICE
        drop:
        - ALL
      runAsGroup: 1000
      runAsUser: 1000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
```

Cheers

**Comment by Mo3m3n** [comment]
Hello @Frankkkkk 
I think we fixed this with the following commit https://github.com/haproxytech/kubernetes-ingress/commit/2fde552bec68719c160c458fee0e68a60b2e94f7
We will release tomorrow or Monday a patch version for v1.7

**Comment by Frankkkkk** [comment]
Hi @Mo3m3n  thanks for the quick reply. I'll try from this commit and I'll keep you up to date.
Thanks and cheers !

**Comment by Mo3m3n** [comment]
This should be fixed in `v1.7.7`
Feel free to reopen this issue if needbe
